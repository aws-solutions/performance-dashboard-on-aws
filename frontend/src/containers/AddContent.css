.usa-form {
  max-width: 28rem !important;
}
