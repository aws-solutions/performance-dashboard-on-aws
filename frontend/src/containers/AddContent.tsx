/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React, { useEffect, useState } from "react";
import { useHistory, useParams } from "react-router-dom";
import { useForm } from "react-hook-form";
import { useTranslation } from "react-i18next";
import { useDashboard, useChangeBackgroundColor } from "../hooks";
import Breadcrumbs from "../components/Breadcrumbs";
import Button from "../components/Button";
import PrimaryActionBar from "../components/PrimaryActionBar";
import RadioButtonsTile from "../components/RadioButtonsTile";
import Alert from "../components/Alert";

interface FormValues {
    widgetType: string;
}

interface PathParams {
    dashboardId: string;
}

function AddContent() {
    const { t } = useTranslation();
    const history = useHistory();
    const { dashboardId } = useParams<PathParams>();
    const { dashboard, loading } = useDashboard(dashboardId);
    const { register, handleSubmit, watch } = useForm<FormValues>();
    const [error, setError] = useState("");

    const widgetType = watch("widgetType");

    useEffect(() => {
        if (error && widgetType) {
            setError("");
        }
    }, [widgetType, error]);

    const onSubmit = async (values: FormValues) => {
        if (values.widgetType === "chart") {
            history.push(`/admin/dashboard/${dashboardId}/add-chart`);
        } else if (values.widgetType === "table") {
            history.push(`/admin/dashboard/${dashboardId}/add-table`);
        } else if (values.widgetType === "text") {
            history.push(`/admin/dashboard/${dashboardId}/add-text`);
        } else if (values.widgetType === "image") {
            history.push(`/admin/dashboard/${dashboardId}/add-image`);
        } else if (values.widgetType === "metrics") {
            history.push(`/admin/dashboard/${dashboardId}/add-metrics`);
        } else if (values.widgetType === "section") {
            history.push(`/admin/dashboard/${dashboardId}/add-section`);
        } else {
            setError(t("AddContentScreen.InvalidWidgetType"));
        }
    };

    const onCancel = () => {
        history.push(`/admin/dashboard/edit/${dashboardId}`);
    };

    const crumbs = [
        {
            label: t("Dashboards"),
            url: "/admin/dashboards",
        },
        {
            label: dashboard?.name,
            url: `/admin/dashboard/edit/${dashboardId}`,
        },
    ];

    useChangeBackgroundColor();

    if (!loading) {
        crumbs.push({
            label: t("AddContentScreen.Title"),
            url: "",
        });
    }

    return (
        <>
            <Breadcrumbs crumbs={crumbs} />
            <div className="grid-row">
                <PrimaryActionBar className="tablet:grid-col-6 grid-col-12">
                    <h1 id="addContentFormHeader" className="margin-top-0">
                        {t("AddContentScreen.Title")}
                    </h1>

                    <form
                        className="usa-form usa-form--large"
                        aria-labelledby="addContentFormHeader"
                        onSubmit={handleSubmit(onSubmit)}
                    >
                        <fieldset className="usa-fieldset">
                            <legend className="margin-y-1 text-semibold display-inline-block font-sans-lg">
                                {t("AddContentScreen.Instructions")}
                            </legend>

                            {error && <Alert type="error" message={error} slim></Alert>}

                            <RadioButtonsTile
                                isHorizontally={false}
                                register={register}
                                options={[
                                    {
                                        id: "text",
                                        value: "text",
                                        name: "widgetType",
                                        dataTestId: "textRadioButton",
                                        label: t("Text"),
                                        description: t("AddContentScreen.TextDescription"),
                                    },
                                    {
                                        id: "metrics",
                                        value: "metrics",
                                        name: "widgetType",
                                        dataTestId: "metricsRadioButton",
                                        label: t("Metrics"),
                                        description: t("AddContentScreen.MetricsDescription"),
                                    },
                                    {
                                        id: "chart",
                                        value: "chart",
                                        name: "widgetType",
                                        dataTestId: "chartRadioButton",
                                        label: t("Chart"),
                                        description: t("AddContentScreen.ChartDescription"),
                                    },
                                    {
                                        id: "table",
                                        value: "table",
                                        name: "widgetType",
                                        dataTestId: "tableRadioButton",
                                        label: t("Table"),
                                        description: t("AddContentScreen.TableDescription"),
                                    },
                                    {
                                        id: "image",
                                        value: "image",
                                        name: "widgetType",
                                        dataTestId: "imageRadioButton",
                                        label: t("Image"),
                                        description: t("AddContentScreen.ImageDescription"),
                                    },
                                    {
                                        id: "section",
                                        value: "section",
                                        name: "widgetType",
                                        dataTestId: "sectionRadioButton",
                                        label: t("Section"),
                                        description: t("AddContentScreen.SectionDescription"),
                                    },
                                ]}
                            />
                        </fieldset>
                        <br />
                        <hr />
                        <Button
                            variant="base"
                            type="submit"
                            disabledToolTip={
                                !widgetType ? t("AddContentScreen.DisabledToolTip") : ""
                            }
                            className="margin-top-1"
                        >
                            {t("ContinueButton")}
                        </Button>
                        <Button
                            variant="unstyled"
                            type="button"
                            onClick={onCancel}
                            className="text-base-dark hover:text-base-darker active:text-base-darkest margin-top-1 "
                        >
                            {t("Cancel")}
                        </Button>
                    </form>
                </PrimaryActionBar>
                <div className="tablet:grid-col-6">
                    {widgetType === "text" && (
                        <img
                            src={`${process.env.PUBLIC_URL}/images/text-content-item.svg`}
                            alt={t("AddContentScreen.TextImageAlt")}
                        />
                    )}

                    {widgetType === "metrics" && (
                        <img
                            src={`${process.env.PUBLIC_URL}/images/metrics-content-item.svg`}
                            alt={t("AddContentScreen.MetricsImageAlt")}
                        />
                    )}

                    {widgetType === "chart" && (
                        <img
                            src={`${process.env.PUBLIC_URL}/images/chart-content-item.svg`}
                            alt={t("AddContentScreen.ChartImageAlt")}
                        />
                    )}

                    {widgetType === "table" && (
                        <img
                            src={`${process.env.PUBLIC_URL}/images/table-content-item.svg`}
                            alt={t("AddContentScreen.TableImageAlt")}
                        />
                    )}

                    {widgetType === "image" && (
                        <img
                            src={`${process.env.PUBLIC_URL}/images/image-content-item.svg`}
                            alt={t("AddContentScreen.ImageAlt")}
                        />
                    )}

                    {widgetType === "section" && (
                        <img
                            src={`${process.env.PUBLIC_URL}/images/section-content-item.svg`}
                            alt={t("AddContentScreen.SectionAlt")}
                        />
                    )}
                </div>
            </div>
        </>
    );
}

export default AddContent;
