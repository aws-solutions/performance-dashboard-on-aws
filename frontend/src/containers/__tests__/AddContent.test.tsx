/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render, fireEvent, act, screen, RenderResult } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { MemoryRouter, Router } from "react-router-dom";
import AddContent from "../AddContent";

describe("Add content screen", () => {
    let component: RenderResult;

    beforeEach(() => {
        component = render(<AddContent />, { wrapper: MemoryRouter });
    });

    test("renders Add content", async () => {
        expect(component).toMatchSnapshot();
    });

    test("renders header", async () => {
        const addContent = await screen.getByRole("heading", {
            name: "Add content item",
        });
        expect(addContent).toBeInTheDocument();
    });

    test("renders text option", async () => {
        const header = await screen.findByText("Text");
        const description = await screen.findByText(
            "Add a formatted block of text. Input supports Markdown including links, bullets, bolding, and headings.",
        );
        expect(header).toBeInTheDocument();
        expect(description).toBeInTheDocument();
    });

    test("renders metrics option", async () => {
        const header = await screen.findByText("Metrics");
        const description = await screen.findByText(
            "Add one or more metrics to show point-in-time numerical data and trends.",
        );
        expect(header).toBeInTheDocument();
        expect(description).toBeInTheDocument();
    });

    test("renders chart option", async () => {
        const header = await screen.findByText("Chart");
        const description = await screen.findByText(
            "Display data as a visualization, including line, bar, column and part-to-whole charts.",
        );
        expect(header).toBeInTheDocument();
        expect(description).toBeInTheDocument();
    });

    test("renders table option", async () => {
        const header = await screen.findByText("Table");
        const description = await screen.findByText("Display data formatted in rows and columns.");
        expect(header).toBeInTheDocument();
        expect(description).toBeInTheDocument();
    });

    test("renders image option", async () => {
        const header = await screen.findByText("Image");
        const description = await screen.findByText("Upload an image to display.");
        expect(header).toBeInTheDocument();
        expect(description).toBeInTheDocument();
    });

    test("renders section option", async () => {
        const header = await screen.findByText("Section");
        const description = await screen.findByText(
            "Add a section to group similar content items in a list.",
        );
        expect(header).toBeInTheDocument();
        expect(description).toBeInTheDocument();
    });

    test("renders preview thumbnail for each content item", async () => {
        fireEvent.click(screen.getByTestId("textRadioButton"));
        let thumbnail = screen.getByAltText("Text Content Item Preview");
        expect(thumbnail).toBeInTheDocument();

        fireEvent.click(screen.getByTestId("metricsRadioButton"));
        thumbnail = screen.getByAltText("Metrics Content Item Preview");
        expect(thumbnail).toBeInTheDocument();

        fireEvent.click(screen.getByTestId("chartRadioButton"));
        thumbnail = screen.getByAltText("Chart Content Item Preview");
        expect(thumbnail).toBeInTheDocument();

        fireEvent.click(screen.getByTestId("tableRadioButton"));
        thumbnail = screen.getByAltText("Table Content Item Preview");
        expect(thumbnail).toBeInTheDocument();

        fireEvent.click(screen.getByTestId("imageRadioButton"));
        thumbnail = screen.getByAltText("Image Content Item Preview");
        expect(thumbnail).toBeInTheDocument();

        fireEvent.click(screen.getByTestId("sectionRadioButton"));
        thumbnail = screen.getByAltText("Section Content Item Preview");
        expect(thumbnail).toBeInTheDocument();
    });
});

describe("'Continue' button navigation", () => {
    let history;
    beforeEach(() => {
        history = createMemoryHistory();
        jest.spyOn(history, "push");
        render(
            <Router history={history}>
                <AddContent />
            </Router>,
        );
    });

    test("takes you to add-chart screen, when Chart option is selected", async () => {
        fireEvent.click(screen.getByTestId("chartRadioButton"));

        await act(async () => {
            const continueButton = await screen.findByRole("button", {
                name: "Continue",
            });
            fireEvent.click(continueButton);
        });

        expect(history.push).toHaveBeenCalledWith("/admin/dashboard/undefined/add-chart");
    });

    test("takes you to add-table screen, when Table option is selected", async () => {
        fireEvent.click(screen.getByTestId("tableRadioButton"));

        await act(async () => {
            const continueButton = await screen.findByRole("button", {
                name: "Continue",
            });
            fireEvent.click(continueButton);
        });

        expect(history.push).toHaveBeenCalledWith("/admin/dashboard/undefined/add-table");
    });

    test("takes you to add-text screen, when Text option is selected", async () => {
        fireEvent.click(screen.getByTestId("textRadioButton"));

        await act(async () => {
            const continueButton = await screen.findByRole("button", {
                name: "Continue",
            });
            fireEvent.click(continueButton);
        });

        expect(history.push).toHaveBeenCalledWith("/admin/dashboard/undefined/add-text");
    });

    test("takes you to add-image screen, when Image option is selected", async () => {
        fireEvent.click(screen.getByTestId("imageRadioButton"));

        await act(async () => {
            const continueButton = await screen.findByRole("button", {
                name: "Continue",
            });
            fireEvent.click(continueButton);
        });

        expect(history.push).toHaveBeenCalledWith("/admin/dashboard/undefined/add-image");
    });

    test("takes you to add-metrics screen, when Metrics option is selected", async () => {
        fireEvent.click(screen.getByTestId("metricsRadioButton"));

        await act(async () => {
            const continueButton = await screen.findByRole("button", {
                name: "Continue",
            });
            fireEvent.click(continueButton);
        });

        expect(history.push).toHaveBeenCalledWith("/admin/dashboard/undefined/add-metrics");
    });

    test("takes you to add-section screen, when Section option is selected", async () => {
        fireEvent.click(screen.getByTestId("sectionRadioButton"));

        await act(async () => {
            const continueButton = await screen.findByRole("button", {
                name: "Continue",
            });
            fireEvent.click(continueButton);
        });

        expect(history.push).toHaveBeenCalledWith("/admin/dashboard/undefined/add-section");
    });
});

test("cancel link takes you to Edit Dashboard screen", async () => {
    const history = createMemoryHistory();
    jest.spyOn(history, "push");

    const { findByRole } = render(
        <Router history={history}>
            <AddContent />
        </Router>,
    );

    await act(async () => {
        const cancelButton = await findByRole("button", { name: "Cancel" });
        fireEvent.click(cancelButton);
    });

    expect(history.push).toHaveBeenCalledWith("/admin/dashboard/edit/undefined");
});
