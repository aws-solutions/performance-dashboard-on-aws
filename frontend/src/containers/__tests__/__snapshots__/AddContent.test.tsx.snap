// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add content screen renders Add content 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="padding-1"
      >
        <nav
          aria-label="Breadcrumbs"
          class="usa-breadcrumb usa-breadcrumb--wrap"
        >
          <ol
            class="usa-breadcrumb__list"
          >
            <li
              class="usa-breadcrumb__list-item"
            >
              <a
                class="usa-breadcrumb__link"
                href="/admin/dashboards"
              >
                <span>
                  Dashboards
                </span>
              </a>
            </li>
            <li
              class="usa-breadcrumb__list-item"
            >
              <a
                class="usa-breadcrumb__link"
                href="/admin/dashboard/edit/undefined"
              >
                <span>
                  My AWS Dashboard
                </span>
              </a>
            </li>
            <li
              aria-current="page"
              class="usa-breadcrumb__list-item usa-current"
            >
              <span>
                Add content item
              </span>
            </li>
          </ol>
        </nav>
      </div>
      <div
        class="grid-row"
      >
        <div
          class="border-base-lighter border-1px shadow-3 z-500 radius-md padding-2 bg-white position-sticky tablet:grid-col-6 grid-col-12"
        >
          <h1
            class="margin-top-0"
            id="addContentFormHeader"
          >
            Add content item
          </h1>
          <form
            aria-labelledby="addContentFormHeader"
            class="usa-form usa-form--large"
          >
            <fieldset
              class="usa-fieldset"
            >
              <legend
                class="margin-y-1 text-semibold display-inline-block font-sans-lg"
              >
                Select the type of content you want to add
              </legend>
              
              <div
                class="padding-2px "
                role="radiogroup"
              >
                <div
                  class="usa-radio"
                  role="radio"
                >
                  <div
                    class="tablet:grid-col"
                  >
                    <input
                      aria-describedby="text-description"
                      class="usa-radio__input usa-radio__input--tile radio"
                      data-testid="textRadioButton"
                      id="text"
                      name="widgetType"
                      type="radio"
                      value="text"
                    />
                    <label
                      class="usa-radio__label"
                      for="text"
                    >
                      Text
                      <span
                        class="text-base usa-prose usa-checkbox__label-description"
                        id="text-description"
                      >
                        Add a formatted block of text. Input supports Markdown including links, bullets, bolding, and headings.
                      </span>
                    </label>
                  </div>
                </div>
                <div
                  class="usa-radio"
                  role="radio"
                >
                  <div
                    class="tablet:grid-col"
                  >
                    <input
                      aria-describedby="metrics-description"
                      class="usa-radio__input usa-radio__input--tile radio"
                      data-testid="metricsRadioButton"
                      id="metrics"
                      name="widgetType"
                      type="radio"
                      value="metrics"
                    />
                    <label
                      class="usa-radio__label"
                      for="metrics"
                    >
                      Metrics
                      <span
                        class="text-base usa-prose usa-checkbox__label-description"
                        id="metrics-description"
                      >
                        Add one or more metrics to show point-in-time numerical data and trends.
                      </span>
                    </label>
                  </div>
                </div>
                <div
                  class="usa-radio"
                  role="radio"
                >
                  <div
                    class="tablet:grid-col"
                  >
                    <input
                      aria-describedby="chart-description"
                      class="usa-radio__input usa-radio__input--tile radio"
                      data-testid="chartRadioButton"
                      id="chart"
                      name="widgetType"
                      type="radio"
                      value="chart"
                    />
                    <label
                      class="usa-radio__label"
                      for="chart"
                    >
                      Chart
                      <span
                        class="text-base usa-prose usa-checkbox__label-description"
                        id="chart-description"
                      >
                        Display data as a visualization, including line, bar, column and part-to-whole charts.
                      </span>
                    </label>
                  </div>
                </div>
                <div
                  class="usa-radio"
                  role="radio"
                >
                  <div
                    class="tablet:grid-col"
                  >
                    <input
                      aria-describedby="table-description"
                      class="usa-radio__input usa-radio__input--tile radio"
                      data-testid="tableRadioButton"
                      id="table"
                      name="widgetType"
                      type="radio"
                      value="table"
                    />
                    <label
                      class="usa-radio__label"
                      for="table"
                    >
                      Table
                      <span
                        class="text-base usa-prose usa-checkbox__label-description"
                        id="table-description"
                      >
                        Display data formatted in rows and columns.
                      </span>
                    </label>
                  </div>
                </div>
                <div
                  class="usa-radio"
                  role="radio"
                >
                  <div
                    class="tablet:grid-col"
                  >
                    <input
                      aria-describedby="image-description"
                      class="usa-radio__input usa-radio__input--tile radio"
                      data-testid="imageRadioButton"
                      id="image"
                      name="widgetType"
                      type="radio"
                      value="image"
                    />
                    <label
                      class="usa-radio__label"
                      for="image"
                    >
                      Image
                      <span
                        class="text-base usa-prose usa-checkbox__label-description"
                        id="image-description"
                      >
                        Upload an image to display.
                      </span>
                    </label>
                  </div>
                </div>
                <div
                  class="usa-radio"
                  role="radio"
                >
                  <div
                    class="tablet:grid-col"
                  >
                    <input
                      aria-describedby="section-description"
                      class="usa-radio__input usa-radio__input--tile radio"
                      data-testid="sectionRadioButton"
                      id="section"
                      name="widgetType"
                      type="radio"
                      value="section"
                    />
                    <label
                      class="usa-radio__label"
                      for="section"
                    >
                      Section
                      <span
                        class="text-base usa-prose usa-checkbox__label-description"
                        id="section-description"
                      >
                        Add a section to group similar content items in a list.
                      </span>
                    </label>
                  </div>
                </div>
              </div>
            </fieldset>
            <br />
            <hr />
            <button
              class="usa-button usa-button--base margin-top-1"
              type="submit"
            >
              Continue
            </button>
            <button
              class="usa-button usa-button--unstyled text-base-dark hover:text-base-darker active:text-base-darkest margin-top-1 "
              type="button"
            >
              Cancel
            </button>
          </form>
        </div>
        <div
          class="tablet:grid-col-6"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="padding-1"
    >
      <nav
        aria-label="Breadcrumbs"
        class="usa-breadcrumb usa-breadcrumb--wrap"
      >
        <ol
          class="usa-breadcrumb__list"
        >
          <li
            class="usa-breadcrumb__list-item"
          >
            <a
              class="usa-breadcrumb__link"
              href="/admin/dashboards"
            >
              <span>
                Dashboards
              </span>
            </a>
          </li>
          <li
            class="usa-breadcrumb__list-item"
          >
            <a
              class="usa-breadcrumb__link"
              href="/admin/dashboard/edit/undefined"
            >
              <span>
                My AWS Dashboard
              </span>
            </a>
          </li>
          <li
            aria-current="page"
            class="usa-breadcrumb__list-item usa-current"
          >
            <span>
              Add content item
            </span>
          </li>
        </ol>
      </nav>
    </div>
    <div
      class="grid-row"
    >
      <div
        class="border-base-lighter border-1px shadow-3 z-500 radius-md padding-2 bg-white position-sticky tablet:grid-col-6 grid-col-12"
      >
        <h1
          class="margin-top-0"
          id="addContentFormHeader"
        >
          Add content item
        </h1>
        <form
          aria-labelledby="addContentFormHeader"
          class="usa-form usa-form--large"
        >
          <fieldset
            class="usa-fieldset"
          >
            <legend
              class="margin-y-1 text-semibold display-inline-block font-sans-lg"
            >
              Select the type of content you want to add
            </legend>
            
            <div
              class="padding-2px "
              role="radiogroup"
            >
              <div
                class="usa-radio"
                role="radio"
              >
                <div
                  class="tablet:grid-col"
                >
                  <input
                    aria-describedby="text-description"
                    class="usa-radio__input usa-radio__input--tile radio"
                    data-testid="textRadioButton"
                    id="text"
                    name="widgetType"
                    type="radio"
                    value="text"
                  />
                  <label
                    class="usa-radio__label"
                    for="text"
                  >
                    Text
                    <span
                      class="text-base usa-prose usa-checkbox__label-description"
                      id="text-description"
                    >
                      Add a formatted block of text. Input supports Markdown including links, bullets, bolding, and headings.
                    </span>
                  </label>
                </div>
              </div>
              <div
                class="usa-radio"
                role="radio"
              >
                <div
                  class="tablet:grid-col"
                >
                  <input
                    aria-describedby="metrics-description"
                    class="usa-radio__input usa-radio__input--tile radio"
                    data-testid="metricsRadioButton"
                    id="metrics"
                    name="widgetType"
                    type="radio"
                    value="metrics"
                  />
                  <label
                    class="usa-radio__label"
                    for="metrics"
                  >
                    Metrics
                    <span
                      class="text-base usa-prose usa-checkbox__label-description"
                      id="metrics-description"
                    >
                      Add one or more metrics to show point-in-time numerical data and trends.
                    </span>
                  </label>
                </div>
              </div>
              <div
                class="usa-radio"
                role="radio"
              >
                <div
                  class="tablet:grid-col"
                >
                  <input
                    aria-describedby="chart-description"
                    class="usa-radio__input usa-radio__input--tile radio"
                    data-testid="chartRadioButton"
                    id="chart"
                    name="widgetType"
                    type="radio"
                    value="chart"
                  />
                  <label
                    class="usa-radio__label"
                    for="chart"
                  >
                    Chart
                    <span
                      class="text-base usa-prose usa-checkbox__label-description"
                      id="chart-description"
                    >
                      Display data as a visualization, including line, bar, column and part-to-whole charts.
                    </span>
                  </label>
                </div>
              </div>
              <div
                class="usa-radio"
                role="radio"
              >
                <div
                  class="tablet:grid-col"
                >
                  <input
                    aria-describedby="table-description"
                    class="usa-radio__input usa-radio__input--tile radio"
                    data-testid="tableRadioButton"
                    id="table"
                    name="widgetType"
                    type="radio"
                    value="table"
                  />
                  <label
                    class="usa-radio__label"
                    for="table"
                  >
                    Table
                    <span
                      class="text-base usa-prose usa-checkbox__label-description"
                      id="table-description"
                    >
                      Display data formatted in rows and columns.
                    </span>
                  </label>
                </div>
              </div>
              <div
                class="usa-radio"
                role="radio"
              >
                <div
                  class="tablet:grid-col"
                >
                  <input
                    aria-describedby="image-description"
                    class="usa-radio__input usa-radio__input--tile radio"
                    data-testid="imageRadioButton"
                    id="image"
                    name="widgetType"
                    type="radio"
                    value="image"
                  />
                  <label
                    class="usa-radio__label"
                    for="image"
                  >
                    Image
                    <span
                      class="text-base usa-prose usa-checkbox__label-description"
                      id="image-description"
                    >
                      Upload an image to display.
                    </span>
                  </label>
                </div>
              </div>
              <div
                class="usa-radio"
                role="radio"
              >
                <div
                  class="tablet:grid-col"
                >
                  <input
                    aria-describedby="section-description"
                    class="usa-radio__input usa-radio__input--tile radio"
                    data-testid="sectionRadioButton"
                    id="section"
                    name="widgetType"
                    type="radio"
                    value="section"
                  />
                  <label
                    class="usa-radio__label"
                    for="section"
                  >
                    Section
                    <span
                      class="text-base usa-prose usa-checkbox__label-description"
                      id="section-description"
                    >
                      Add a section to group similar content items in a list.
                    </span>
                  </label>
                </div>
              </div>
            </div>
          </fieldset>
          <br />
          <hr />
          <button
            class="usa-button usa-button--base margin-top-1"
            type="submit"
          >
            Continue
          </button>
          <button
            class="usa-button usa-button--unstyled text-base-dark hover:text-base-darker active:text-base-darkest margin-top-1 "
            type="button"
          >
            Cancel
          </button>
        </form>
      </div>
      <div
        class="tablet:grid-col-6"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
